src/block_cache/block_chunkcache.c

```diff
@@ -916,6 +916,52 @@ __wt_chunkcache_get(WT_SESSION_IMPL *session, WT_BLOCK *block, uint32_t objectid
     return (0);
 }
 
+/*
+ * __wt_chunkcache_free_external --
+ *     Find chunks in the chunk cache using object id, and free the chunks.
+ */
+int
+__wt_chunkcache_free_external(
+  WT_SESSION_IMPL *session, WT_BLOCK *block, uint32_t objectid, wt_off_t offset, uint32_t size)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
+    WT_CHUNKCACHE_HASHID hash_id;
+    size_t already_read;
+    uint64_t bucket_id;
+    const char *object_name;
+    chunkcache = &S2C(session)->chunkcache;
+    already_read = 0;
+    object_name = NULL;
+
+    /* Only cache read-only tiered objects. */
+    if (!block->readonly)
+        return (0);
+
+    WT_RET(
+      __wt_tiered_name(session, session->dhandle, 0, WT_TIERED_NAME_SKIP_PREFIX, &object_name));
+
+    /* A block may span multiple chunks, loop until we have read all the data. */
+    while (already_read < size) {
+        bucket_id = __chunkcache_tmp_hash(
+          chunkcache, &hash_id, object_name, objectid, offset + (wt_off_t)already_read);
+        __wt_spin_lock(session, WT_BUCKET_LOCK(chunkcache, bucket_id));
+
+        /* If a chunk is matched, remove it from the bucket queue and free the chunk. */
```

Comment:
A couple of nitpicky things here - (1) comments should describe why the code does something, rather than what it does. And (2) can the spinlock be moved down into the same "paragraph" as the unlock please? It's important to match these and having them visually line up helps with checking that.

Reply:
Yup I have addressed (2) here: [6d1b1da](https://github.com/wiredtiger/wiredtiger/pull/9838/commits/6d1b1da34e6d1213840588e9088bfc8f1c674881). 
Regarding (1) - I like adding descriptive comments to make sections of the code more readable and understandable.

Reply:
Fair enough - thanks.